I. T-Test

1. Objective
Compare the mean closing price (close) between two groups:  
Group 1: Closing prices before 2022.  
Group 2: Closing prices after 2022.  

2.  Hypotheses  
Null hypothesis (H0): There is no difference in the mean closing price between the two groups.   
Alternative hypothesis (H1): There is a significant difference in the mean closing price between the two groups.

3. Prepare data


In [1]:
import pandas as pd
from scipy.stats import ttest_ind

# Đọc dữ liệu
file_path = 'E:/Code/STAT3013.-P12_Nhom4/Dataset/Platinum_data_filtered.xlsx'
df = pd.read_excel(file_path)

# Chuyển đổi cột 'date' thành định dạng datetime
df['date'] = pd.to_datetime(df['date'])

# Chia dữ liệu thành hai nhóm: trước và sau ngày 2020-01-01
group1 = df[df['date'] < '2022-01-01']['close']  # Trước ngày 01/01/2022
group2 = df[df['date'] >= '2022-01-01']['close']  # Sau ngày 01/01/2022
# Kiểm tra thông tin chi tiết về DataFrame, bao gồm kiểu dữ liệu của các cột



4. Perform T-Test

In [2]:
# Thực hiện kiểm định t-test
t_stat, p_value = ttest_ind(group1, group2, equal_var=False)  # equal_var=False khi phương sai hai nhóm không bằng nhau

# In kết quả
print("T-statistic:", t_stat)
print("P-value:", p_value)


T-statistic: -1.5390856149264587
P-value: 0.12407824206464654


5. Conclusion


p = 0.124 > 0.05: There is no difference in the mean closing price between the two groups.

The mean values of the two groups are likely to be similar, and any observed difference could be due to random variation rather than a true difference.

II. Chi-square test

1. Objective  
Examine the relationship between two categorical variables:  
close_category: Classified as High or Low based on the mean closing price.  
volume_category: Classified as High or Low based on the mean trading volume.

2. Hypotheses  
Null hypothesis (H0): There is no relationship between closing price category and trading volume category.  
Alternative hypothesis (H1): There is a significant relationship between closing price category and trading volume category.

In [3]:
# Phân loại giá đóng cửa (close) và khối lượng giao dịch (volume) thành "High" và "Low"
mean_close = df['close'].mean()
mean_volume = df['volume'].mean()

df['close_category'] = ['High' if close >= mean_close else 'Low' for close in df['close']]
df['volume_category'] = ['High' if volume >= mean_volume else 'Low' for volume in df['volume']]

In [4]:
# Tạo bảng tần suất chéo
contingency_table = pd.crosstab(df['close_category'], df['volume_category'])

# Hiển thị bảng tần suất
print(contingency_table)


volume_category  High  Low
close_category            
High               30  567
Low                32  698


In [5]:
from scipy.stats import chi2_contingency

# Thực hiện kiểm định Chi-square
chi2_stat, p_value, dof, expected = chi2_contingency(contingency_table, correction=False)

# In kết quả
print("Chi-square statistic:", chi2_stat)
print("P-value:", p_value)
print("Degrees of freedom:", dof)
print("Expected frequencies:")
print(expected)


Chi-square statistic: 0.30350500044055756
P-value: 0.5816933982160836
Degrees of freedom: 1
Expected frequencies:
[[ 27.89299171 569.10700829]
 [ 34.10700829 695.89299171]]


Conclusion

P-value = 0.582: (>0.05), there is no statistical evidence to conclude that there is a relationship between close_category and volume_category. This means that "close price high/low" and "high/low trading volume" may not be related in this dataset.

III. Build a confidence gap

In [6]:
import numpy as np
import pandas as pd
from scipy.stats import norm


# Lấy cột giá đóng cửa (close)
data = df['close']


In [7]:
# Tính trung bình, độ lệch chuẩn, và cỡ mẫu
mean = np.mean(data)
std_dev = np.std(data, ddof=1)  # ddof=1 để tính độ lệch chuẩn mẫu
n = len(data)


In [8]:
# Giá trị Z cho mức tin cậy 95% (1.96)
z = norm.ppf(0.975)  # norm.ppf(0.975) = 1.96 cho phân phối chuẩn

# Tính khoảng tin cậy
margin_of_error = z * (std_dev / np.sqrt(n))
confidence_interval = (mean - margin_of_error, mean + margin_of_error)

# In kết quả
print("Mean:", mean)
print("Standard Deviation:", std_dev)
print("Sample Size:", n)
print("Confidence Interval:", confidence_interval)


Mean: 956.0891468361435
Standard Deviation: 103.71034995772727
Sample Size: 1327
Confidence Interval: (np.float64(950.5091400214915), np.float64(961.6691536507956))


Conclusion

95% confidence interval = [950.509, 961.670]